bounce.py

```
@@ -23,21 +22,82 @@
     redirect_url = bounce_url
-    if validators.domain(domain):
-        extracted = tldextract.extract(domain)
-        domain = f"{extracted.domain}.{extracted.suffix}"
-        logging.info(f"Valid domain extracted: {domain}")
-        domain_counter_stat.labels(domain).inc()
-        redirect_url = bounce_redirect_url + domain
+    sanitized_domain = sanitize_domain(domain)
+    extracted_domain = extract_domain(sanitized_domain)
+
+    if extracted_domain:
+        logging.info(f"Valid domain extracted: {extracted_domain}")
```

Ruff

Logging statement uses f-string

(G004)